lib/subscribers.rb

```
@@ -0,0 +1,33 @@
+class Subscribers
+  def self.notify(doi, source, previous_total, current_total)
+    return unless current_total > previous_total
+    low_end = previous_total + 1
+    range = (low_end..current_total)
+    journal = journal_key(doi)
+    subs = get(journal, source)
+    subs.each do |s|
+      milestones = s[:milestones]
+      # returns the last milestone in range
+      hit = milestones.select{ |m| range.include?(m) }.last
+      if hit
+        Faraday.get(s[:url], doi: doi, milestone: hit)
+      end
+    end
+  end
+  
+  def self.journal_key(doi)
+    prefix = '10.1371/journal.'
+    return unless doi && doi.start_with?(prefix)
+    doi.sub(prefix, '').split('.').first
+  end
+
+  def self.get(journal, source)
+    subs = get_from_config
+    return subs.select { |s| s.values_at(:journal, :source) == [journal, source] }
+  end
+
+  def self.get_from_config
+    config = EnvConfig.config_for("SUBSCRIBERS__")
```

WDYTA doing this eagerly (at server start) rather than every time `Subscribers.notify` is called. I understand that as-is, it's nice that we don't need to reload the server if the env var changes, but it's a kind of expensive and potentially error-prone thing to do everytime a source updates for an article. I'd prefer to know that there's a config problem at boot, rather than seeing 20k exceptions later that day.

---

You describe a great reason to have an initializer.  I deleted it when I should have refactored it.

---

@slimeate updated to fail early: https://github.com/PLOS/lagotto/pull/44/commits/4b36167058307daf8249f0ccda42261d1854baf1